addons/website_sale/models/product_template.py

```diff
@@ -421,3 +422,13 @@ def get_google_analytics_data(self, combination):
             'currency': product.currency_id.name,
             'price': combination['list_price'],
         }
+
+    def _get_contextual_pricelist(self):
+        """ Override to fallback on website current pricelist
+        """
+        # YTI TODO: During website_sale cleaning, we should get rid of those crappy context thing
+        pricelist = super()._get_contextual_pricelist()
+        if pricelist:
+            return pricelist
+        website = ir_http.get_request_website()
+        return website and website.get_current_pricelist()
```

Comment:
```suggestion
        if website:
            return website.get_current_pricelist()
        return pricelist
```

Make sure the type of the returned value is consistent, it might be False here if no website was found.